### PR TITLE
Release v0.5.0

### PR DESCRIPTION
**Pull request by gadomski**
**Description:**

Release v0.5.0. Not all issues in the v0.5.0 milestone have been closed, but the API refactors and deprecations from #284 are significant enough (IMO) to warrant their own MAJOR release (we're still in v0.y.z, so MAJOR releases bump the second number). If approved, we should move all remaining tickets in https://github.com/stac-utils/pystac-client/milestone/5 to https://github.com/stac-utils/pystac-client/milestone/8.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

## Annotated tag contents

```
v0.5.0

Added:

- Added a new keyword `modifier` to various constructors like `Client.open()` [#259](https://github.com/stac-utils/pystac-client/issues/259)

Fixed:

- Fix type annotation of `Client._stac_io` and avoid implicit re-exports in `pystac_client.__init__.py` [#249](https://github.com/stac-utils/pystac-client/pull/249)
- Added `ItemSearch.pages`, `ItemSearch.pages_as_dicts`, `ItemSearch.item_collection`, and `ItemSearch.item_collection_as_dict`
  as replacements for various deprecated methods [#237](https://github.com/stac-utils/pystac-client/issues/237)
- Restored the previous behavior of ``Client.search()`` to return an unlimited number of items by default. [#273](https://github.com/stac-utils/pystac-client/pull/273)

Deprecated:

- `ItemSearch.item_collections` has been deprecated in favor of `ItemSearch.pages`. [#237](https://github.com/stac-utils/pystac-client/issues/237)
```

### CODE REVIEW

**Comment by gadomski** [comment]
Converting to draft because I need to update the front page of the docs to include this new v0.5.0 version.

**Comment by gadomski** [comment]
Ok updated the table, ready for review.

**Review by TomAugspurger — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,11 +58,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Search parameter `intersects` is now typed to only accept a str, dict, or object that implements `__geo_interface__` [#169](https://github.com/stac-utils/pystac-client/pull/169)
```
[the remark this answers is not in the document]
Comment (gadomski) [reply CHANGELOG.md]: Good catch, thanks. Fixed.

**Review by matthewhanson — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -60,7 +72,6 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   `item_collections()` should be used instead. [#206](https://github.com/stac-utils/pystac-client/pull/206)
 - CLI parameter `-q` is now deprecated and may be removed as early as v0.5.0. Use `--query` instead. [#215](https://github.com/stac-utils/pystac-client/pull/215)
 
-
 ## Removed
 
 - Client parameter `require_geojson_link` has been removed. [#169](https://github.com/stac-utils/pystac-client/pull/169)
```
[the remark this answers is not in the document]
Comment (matthewhanson) [reply CHANGELOG.md:77]: comment on wrong line - PR for `require_geojson_link` should be PR #165

[the remark this answers is not in the document]
Comment (gadomski) [reply CHANGELOG.md]: thanks fixed.

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**